test/functional/feature_assumeutxo.py

```diff
@@ -218,6 +222,27 @@ def test_snapshot_block_invalidated(self, dump_output_path):
             assert_raises_rpc_error(-32603, msg, node.loadtxoutset, dump_output_path)
             node.reconsiderblock(block_hash)
 
+    def test_snapshot_not_on_most_work_chain(self, dump_output_path):
+        self.log.info("Test bitcoind should fail when the node knows the headers of another chain with more work.")
+        node0 = self.nodes[0]
+        node1 = self.nodes[1]
+        # create two blocks that form a longer chain
+        parent = node0.getblockhash(SNAPSHOT_BASE_HEIGHT - 1)
+        block_time = node0.getblock(node0.getbestblockhash())['time'] + 1
+        fork1 = create_block(int(parent, 16), create_coinbase(SNAPSHOT_BASE_HEIGHT - 1), block_time)
```

Comment:
Huh, either I am confused or something isn't right here: The `parent` above is at height `SNAPSHOT_BASE_HEIGHT - 1` and now the child of this `parent` has a coinbase also at height `SNAPSHOT_BASE_HEIGHT - 1`?  It seems this isn't validated so it's not a huge issue though.